software/base/src/main/java/org/apache/brooklyn/entity/software/base/lifecycle/MachineLifecycleEffectorTasks.java

```diff
@@ -709,7 +712,26 @@ protected void doStop(ConfigBag parameters, Callable<StopMachineDetails<Integer>
 
         DynamicTasks.queue("pre-stop", new PreStopCustomTask());
 
+        // BROOKLYN-263:
+        // With this change the stop effector will wait for Location to provision so it can stop it.
+        // It will not retry if non-jclouds location is used in the entity.
         Maybe<MachineLocation> machine = Machines.findUniqueMachineLocation(entity().getLocations());
+        if (Boolean.TRUE.equals(entity().sensors().get(Attributes.JCLOUDS_PROVISIONING_STARTED))
+                && machine.isAbsent()) {
+            entity().sensors().set(Attributes.JCLOUDS_PROVISIONING_STARTED, null);
```

Comment:
@aledsage do you have an idea how can I tell the stop task that the provisioning started other than the sensor flag. I feel like the code will need serious refactoring in order stop task to know earlier about the jclouds node.
The entity knows only about the machine location. It doesn't know about the LocationProvisioner.